OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnTerrain.cs

```diff
@@ -55,10 +55,11 @@ void ITick.Tick(Actor self)
 			var wantsGranted = info.TerrainTypes.Contains(currentTerrain);
 			if (currentTerrain != previousTerrain)
 			{
-				if (wantsGranted && conditionToken == ConditionManager.InvalidConditionToken)
-					conditionToken = conditionManager.GrantCondition(self, info.Condition);
-				else if (!wantsGranted && conditionToken != ConditionManager.InvalidConditionToken)
-					conditionToken = conditionManager.RevokeCondition(self, conditionToken);
+				if (self.Location.Layer == 0)
```

Comment:
Maby the Layer check is only needed for the GrandCondition part.

Reply:
This should move up into the `wantsGranted` definition.

Reply:
Done